src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,177 @@
+        } else {
+            return someDate + " is today";
+        }
     }
 
     /**
      * Given a String representation of a date and some timezone,
      * return LocalDateTime in this timezone.
      */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateInString);
+        LocalDateTime localDateTime = zonedDateTime.toLocalDateTime();
+        ZoneId zoneId = ZoneId.of(zone);
+        ZoneOffset zoneOffset = zoneId.getRules().getOffset(localDateTime);
+        return LocalDateTime.ofInstant(Instant.parse(dateInString), zoneOffset);
     }
 
     /**
      * Given some LocalDateTime, return an OffsetDateTime with the local time offset applied
      * (`+02:00` for Ukraine).
-     *
+     * <p>
      * Example: we receive a LocalDateTime with a value `2019-09-06T13:17`.
-     *          We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
-     *          where `+02:00` is the offset for our local timezone.
-     *
+     * We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
+     * where `+02:00` is the offset for our local timezone.
+     * <p>
      * OffsetDateTime is recommended to use for storing date values in a database.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return localTime.atOffset(ZoneOffset.of(TIME_ZONE));
     }
 
     /**
      * Given a String formatted as `yyyymmdd`,
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        return (Integer.parseInt(date.substring(START_MONTH_POSITION, END_MONTH_POSITION))
+                > MAX_MONTH) ? Optional.empty()
+                : Optional.ofNullable(LocalDate.from(DateTimeFormatter.BASIC_ISO_DATE.parse(date)));
     }
 
     /**
      * Given a String formatted as `d MMM yyyy` (MMM - Sep, Oct, etc),
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> customParseDate(String date) {
-        return Optional.empty();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d MMM yyyy", locale);
+        return (Integer.parseInt(date.substring(START_DAY_POSITION, END_DAY_POSITION))
+                > MAX_DAY_IN_MONTH) ? Optional.empty()
+                : Optional.ofNullable(LocalDate.from(formatter.parse(date)));
     }
 
     /**
      * Given some LocalDateTime, return a String formatted as
      * `day(2-digit) month(full name in English) year(4-digit) hours(24-hour format):minutes`.
-     *
+     * <p>
      * Example: "01 January 2000 18:00".
      */
     public String formatDate(LocalDateTime dateTime) {
-        return "";
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMMM yyyy HH:mm", locale);
```

Comment:
let's create a constant fo this formatter

Reply:
done